azure/services/routetables/client.go

```diff
@@ -18,138 +18,103 @@ package routetables
 	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
 // azureClient contains the Azure go-sdk Client.
 type azureClient struct {
-	routetables network.RouteTablesClient
+	routetables *armnetwork.RouteTablesClient
 }
 
-// newClient creates a new route tables client from subscription ID.
-func newClient(auth azure.Authorizer) *azureClient {
-	c := newRouteTablesClient(auth.SubscriptionID(), auth.BaseURI(), auth.Authorizer())
-	return &azureClient{c}
-}
-
-// newRouteTablesClient creates a new route tables client from subscription ID.
-func newRouteTablesClient(subscriptionID string, baseURI string, authorizer autorest.Authorizer) network.RouteTablesClient {
-	routeTablesClient := network.NewRouteTablesClientWithBaseURI(baseURI, subscriptionID)
-	azure.SetAutoRestClientDefaults(&routeTablesClient.Client, authorizer)
-	return routeTablesClient
+// newClient creates a new route tables client from an authorizer.
+func newClient(auth azure.Authorizer) (*azureClient, error) {
+	opts, err := azure.ARMClientOptions(auth.CloudEnvironment())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create routetables client options")
+	}
+	factory, err := armnetwork.NewClientFactory(auth.SubscriptionID(), auth.Token(), opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create armnetwork client factory")
+	}
+	return &azureClient{factory.NewRouteTablesClient()}, nil
 }
 
 // Get gets the specified route table.
 func (ac *azureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (result interface{}, err error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "routetables.azureClient.Get")
 	defer done()
 
-	return ac.routetables.Get(ctx, spec.ResourceGroupName(), spec.ResourceName(), "")
+	resp, err := ac.routetables.Get(ctx, spec.ResourceGroupName(), spec.ResourceName(), nil)
+	if err != nil {
+		return nil, err
+	}
+	return resp.RouteTable, nil
 }
 
 // CreateOrUpdateAsync creates or updates a route table asynchronously.
-// It sends a PUT request to Azure and if accepted without error, the func will return a Future which can be used to track the ongoing
+// It sends a PUT request to Azure and if accepted without error, the func will return a Poller which can be used to track the ongoing
 // progress of the operation.
-func (ac *azureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter, parameters interface{}) (result interface{}, future azureautorest.FutureAPI, err error) {
+func (ac *azureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter, resumeToken string, parameters interface{}) (result interface{}, poller *runtime.Poller[armnetwork.RouteTablesClientCreateOrUpdateResponse], err error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "routetables.azureClient.CreateOrUpdateAsync")
 	defer done()
 
-	rt, ok := parameters.(network.RouteTable)
+	rt, ok := parameters.(armnetwork.RouteTable)
 	if !ok {
```

Comment:
Should we add `&& parameters != nil` here, anticipating #3934?

Reply:
Yeah, that's probably easier than doing it after that PR lands.